src/scaladoc/scala/tools/nsc/ScalaDoc.scala

```diff
@@ -19,13 +19,15 @@ class ScalaDoc {
 
   def process(args: Array[String]): Boolean = {
     var reporter: ScalaDocReporter = null
-    val docSettings = new doc.Settings(msg => reporter.error(FakePos("scaladoc"), msg + "\n  scaladoc -help  gives more information"),
+    val docSettings = new doc.Settings(msg => reporter.error(FakePos("scaladoc"), msg + "\n  scaladoc -help gives more information"),
                                        msg => reporter.printMessage(msg))
     reporter = new ScalaDocReporter(docSettings)
     val command = new ScalaDoc.Command(args.toList, docSettings)
     def hasFiles = command.files.nonEmpty || docSettings.uncompilableFiles.nonEmpty
 
-    if (docSettings.version.value)
+    if (!command.ok)
+      return false; // hard exit if there are errors, e.g. bad arguments
```

Comment:
The minimal change would be to replace line 29 with (), but that looks weird to me. [See changes below](https://github.com/scala/scala/pull/5215/files/160b95faedb84fb517c9ca4519e197e7bc847948..956a3a20ec7c10561b4f7db42578c2ebed6f9e57) for an alternative that changes the output, but reads a little better:

```
chase@chase-kubuntu:~/projects/scala$ ./build/quick/bin/scaladoc -h
scaladoc error: bad option: '-h'
Usage: scaladoc <options> <source files>
scaladoc -help gives more information
```

I can change it to whichever is preferred.

Also, I know PRs are supposed to be one commit -- should I submit a new PR with only one commit once the changes look good? I'm not familiar with the PR workflow.